x-pack/filebeat/input/awss3/input.go

```diff
@@ -410,5 +413,36 @@ func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
 	}
 }
 
+func pollSqsUtilizationMetric(ctx context.Context, receiver *sqsReader, metricReporterChan chan int64) {
+	defer close(metricReporterChan)
+
+	t := time.NewTicker(5 * time.Second)
+	defer t.Stop()
+
+	var utilizedNanos int64
+	go func() {
+		for elem := range metricReporterChan {
+			atomic.AddInt64(&utilizedNanos, elem)
+		}
+	}()
+
+	lastTick := time.Now()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case tick := <-t.C:
+			denom := float64(tick.Sub(lastTick)) * float64(receiver.maxMessagesInflight)
+
+			utilizedRate := float64(utilizedNanos) / denom
+			receiver.metrics.sqsWorkerUtilization.Set(utilizedRate)
+
+			// reset for the next polling duration
+			utilizedNanos = 0
+			lastTick = tick
```

Comment:
```suggestion
			utilizedRate := float64(atomic.SwapInt64(&utilizedNanos, 0)) / denom
			receiver.metrics.sqsWorkerUtilization.Set(utilizedRate)

			// reset for the next polling duration
			lastTick = tick
```
(if a value is accessed via an atomic operation, all concurrent accesses of that value must be via atomic operations — also this prevents interleaving between an addition in the `metricReporterCan` receive loop from being clobbered here).